tests/cli/test_status.py

```diff
@@ -503,3 +512,119 @@ def test_call_cli_networkd_masked(self, is_masked_mock, is_active_mock):
+    @patch('netplan_cli.cli.state.SystemConfigState.query_resolved')
+    @patch('netplan_cli.cli.state.SystemConfigState.resolvconf_json')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_online_state')
+    def test_call_cli_diff_shallow(self, online_mock, resolvconf_mock, rd_mock, routes_mock,
+                                   nm_mock, networkd_mock, iproute2_mock, systemctl_mock,
+                                   diff_state_get_diff_mock, diff_state_init_mock, state_init_mock):
+        systemctl_mock.return_value = None
+        iproute2_mock.return_value = [FAKE_DEV]
+        nm_mock.return_value = []
+        routes_mock.return_value = (None, None)
+        rd_mock.return_value = (None, None)
+        resolvconf_mock.return_value = {'addresses': [], 'search': [], 'mode': None}
+        online_mock.return_value = False
+        state_init_mock.return_value = None
+        diff_state_init_mock.return_value = None
+        diff_state_get_diff_mock.return_value = {}
+        state = SystemConfigState()
+        networkd_mock.return_value = state.process_networkd(NETWORKD)
+        out = self._call(['--diff'])
+        self.assertIn('Use "--diff-only" to omit the information that is consistent', out)
+
+    @patch('netplan_cli.cli.state.NetplanConfigState.__init__')
+    @patch('netplan_cli.cli.state_diff.NetplanDiffState.__init__')
+    @patch('netplan_cli.cli.state_diff.NetplanDiffState.get_diff')
+    @patch('netplan_cli.cli.utils.systemctl')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_iproute2')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_networkd')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_nm')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_routes')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_resolved')
+    @patch('netplan_cli.cli.state.SystemConfigState.resolvconf_json')
+    @patch('netplan_cli.cli.state.SystemConfigState.query_online_state')
+    def test_call_cli_diff_only_shallow(self, online_mock, resolvconf_mock, rd_mock, routes_mock,
+                                        nm_mock, networkd_mock, iproute2_mock, systemctl_mock,
+                                        diff_state_get_diff_mock, diff_state_init_mock, state_init_mock):
+        systemctl_mock.return_value = None
+        iproute2_mock.return_value = [FAKE_DEV]
+        nm_mock.return_value = []
+        routes_mock.return_value = (None, None)
+        rd_mock.return_value = (None, None)
+        resolvconf_mock.return_value = {'addresses': [], 'search': [], 'mode': None}
+        online_mock.return_value = False
+        state_init_mock.return_value = None
+        diff_state_init_mock.return_value = None
+        diff_state_get_diff_mock.return_value = {}
+        state = SystemConfigState()
+        networkd_mock.return_value = state.process_networkd(NETWORKD)
+        out = self._call(['--diff-only'])
+        self.assertEqual('', out)
```

Comment:
question (non-blocking): Should we put some message in case where there isn't any diff to show?

Reply:
Agreed. That also seems to be what `git diff` does.